File: namastack-outbox-examples/namastack-outbox-example-java/src/main/java/io/namastack/demo/customer/CustomerRegisteredEvent.java
package io.namastack.demo.customer;

import java.util.UUID;

public class CustomerRegisteredEvent {

  private final UUID id;
  private final String firstname;
  private final String lastname;
  private final String email;

  public CustomerRegisteredEvent(UUID id, String firstname, String lastname, String email) {
    this.id = id;
    this.firstname = firstname;
    this.lastname = lastname;
    this.email = email;
  }

  public UUID getId() {
    return id;
  }

  public String getFirstname() {
    return firstname;
  }

  public String getLastname() {
    return lastname;
  }

  public String getEmail() {
    return email;
  }
}

File: namastack-outbox-examples/namastack-outbox-example-multicaster/src/main/kotlin/io/namastack/demo/customer/CustomerRegisteredEvent.kt
package io.namastack.demo.customer

import io.namastack.outbox.annotation.OutboxEvent
import java.util.UUID

@OutboxEvent(key = "#root.id.toString()")
data class CustomerRegisteredEvent(
    val id: UUID,
    val firstname: String,
    val lastname: String,
    val email: String,
)
